flake.nix

```diff
@@ -7,11 +7,77 @@
 
   outputs = { self, nixpkgs }: {
 
-    lib = {
-      getHashSubdomain = nixosCfg :
-        let prefixLength = nixpkgs.lib.stringLength "/nix/store/";
-            hash = builtins.substring prefixLength 32 nixosCfg.config.system.build.toplevel.drvPath;
-        in hash + ".hash.garnix.me";
+    nixosModules.garnix = { lib, config, ... }:
+      let
+        cfg = config.garnix.persistence;
+      in
+      {
+
+        options.garnix.persistence = {
+          enable = lib.mkEnableOption "Enable persistence in garnix deploys";
+
+          name = lib.mkOption {
+            type = lib.types.str;
+            description = "A unique name to identify this persistent nixos configuration. If a subsequent deploy defines a server with the same persistence name, it'll reuse the same machine, including disks.";
+          };
+        };
```

Comment:
Do we have some documentation for persistent nodes somewhere on our website? If yes, we should probably add a link somewhere here. Either just in a comment or maybe even in one of the `description`s?

TIOLI

Reply:
Not yet, because we don't "really" support it yet and we paused development for it, at least until we're done with the current iteration of monitoring and pr deployments.